scheduler/server/stateful_scheduler.go

```diff
@@ -292,7 +297,15 @@ func NewStatefulScheduler(
 		stat:             stat,
 
 		tasksByJobClassAndStartTimeSec: tasksByClassAndStartMap,
+		persistor:                      persistor,
+	}
+
+	sched.setThrottle(-1)
+
+	if sched.persistor == nil {
```

Comment:
nit: seems weird to create this statefulScheduler and then check if one of the fields we assigned it is nil... could we check if persistor is nil above the statefulScheduler instantiation?

Reply:
actually, this is the 'style' I'm trying to get to... use nopPersistor as default behavior - allow so so consumers of statefulScheduler don't have to think/know about Persistor if they don't care about that feature.